stable/ambassador/templates/deployment.yaml

```diff
@@ -12,6 +12,9 @@ metadata:
     helm.sh/chart: {{ include "ambassador.chart" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- if .Values.extraLabels }}
+      {{- toYaml .Values.extraLabels | nindent 4 }}
+    {{- end }}
```

Comment:
to be consistent with how other charts seems to use `extraLabels` I believe these should be added to all resources. 
Think I'd rather add `deployment.labels` option to set extra labels for the deployment.

```
deployment:
  labels: {}
```

to allow adding labels to the deployment

or we can add the `extraLabels` but set it on all resources while keeping `podLabels` for backwards compat for now.